tests/config.tests/prep.json

```diff
@@ -0,0 +1,25 @@
+{
+  "version": 1,
+  "blobset": {
+    "blobs": {
+      "fw_payload": {
+        "phase": "prep",
+        "type": "base64",
+        "content_type": "application/zip",
+        "payload": "UEsDBAoAAAAAAAZZ3VSU7ZiuBQAAAAUAAAAIABwAYm9vdC5pbWdVVAkAA9xqvGLcarxidXgLAAEEFjQAAARTXwEAYm9vdApQSwMECgAAAAAAB1ndVK9osDMDAAAAAwAAAAYAHABmdy5pbWdVVAkAA95qvGLearxidXgLAAEEFjQAAARTXwEAZncKUEsBAh4DCgAAAAAABlndVJTtmK4FAAAABQAAAAgAGAAAAAAAAQAAAKSBAAAAAGJvb3QuaW1nVVQFAAPcarxidXgLAAEEFjQAAARTXwEAUEsBAh4DCgAAAAAAB1ndVK9osDMDAAAAAwAAAAYAGAAAAAAAAQAAAKSBRwAAAGZ3LmltZ1VUBQAD3mq8YnV4CwABBBY0AAAEU18BAFBLBQYAAAAAAgACAJoAAACKAAAAAAA="
+      },
+      "update_payload": {
+        "phase": "prep",
+        "type": "url",
```

Comment:
Looking at this, are the type and href/payload field redundant?  That is, could it be specified as either:
1) _type_ and _src_ fields -- where the _type_ says what to do with the _src_?  So if _type_ is "base64" then it's inline, if it's "url" then URL, etc...
2) One of { _base64_, _href_ } with no _type_ -- so if _base64_ is defined then it's that, if it's _href_ then URL, etc... It would be an error to have multiple fields defined.

Reply:
Unless if it were expensive to have multiple fields, I think using explicitly named ones is better. Also It's much more straightforward to validate a field named url in one pass, instead of branching based on type etc. 

Alternatively I think you are saying it needs to be an error for type=url, href="...defined" and also payload!=null, but similarly it needs to be an error if type=url and payload!=Url_format, if we are doing error checking.